tensorflow/core/kernels/cwise_op_abs.cc

```diff
@@ -37,8 +37,16 @@ REGISTER_KERNEL_BUILDER(Name("Abs")
                         UnaryOp<CPUDevice, functor::abs<int32>>);
 #endif
 
-#if TENSORFLOW_USE_SYCL
-REGISTER3(UnaryOp, SYCL, "Abs", functor::abs, float, double, int64);
+#ifdef TENSORFLOW_USE_SYCL
+#define REGISTER_SYCL_CWISE_KERNEL(type)                          \
```

Comment:
For the cwise ops, you could use the following (from https://github.com/tensorflow/tensorflow/pull/11234#discussion_r126401862) instead of explicitly creating a kernel builder for each type. I guess it doesn't really make much difference though.
```
#define REGISTER_SYCL_KERNEL(type)                                  \
  REGISTER(BinaryOp, SYCL, "Add", functor::add, type);
TF_CALL_SYCL_NUMBER_TYPES(REGISTER_SYCL_KERNEL);
```